core/types/transaction_signing.go

```diff
@@ -112,6 +112,9 @@ type EIP155Signer struct {
 }
 
 func NewEIP155Signer(chainId *big.Int) EIP155Signer {
+	if chainId == nil {
+		chainId = new(big.Int)
+	}
```

Comment:
I'm not sure about this change. To me it feels like it's hiding programming bugs with silent chain id assignment. I'd much rather add a check when initializing the blockchain (or somewhere around that level) that errors out if chainid is not set. This feels like a time bomb waiting for a refactor to blow up.